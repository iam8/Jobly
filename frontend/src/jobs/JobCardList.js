import React from "react";

import JobCard from "./JobCard";


/** Render a list of JobCard components. Each JobCard displays basic info about a job. */
function JobCardList({jobList}) {

    return (
        <div className="JobCardList">
            {jobList.map((job) => {
                return <div className="mb-3" key={job.id}>
                    <JobCard
                        id={job.id}
                        title={job.title}
                        salary={job.salary}
                        equity={job.equity}
                        companyName={job.companyName}
                    />
                </div>
            })}
        </div>
    )
}


export default JobCardList;
