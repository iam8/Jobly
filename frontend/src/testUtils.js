import React from "react";
import UserContext from "./auth/UserContext";


const demoUser = {
    username: "testuser",
    firstName: "testfirst",
    lastName: "testlast",
    email: "test@test.net",
};


const DemoUserProvider = ({ children, currentUser = demoUser, hasAppliedToJob = () => false }) => (
    <UserContext.Provider value={{ currentUser, hasAppliedToJob }}>
        {children}
    </UserContext.Provider>
);


export { DemoUserProvider };
