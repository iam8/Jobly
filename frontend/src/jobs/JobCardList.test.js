import React from "react";
import { render } from "@testing-library/react";
import { DemoUserProvider as UserProvider } from "../testUtils";
import JobCardList from "./JobCardList";


test("Renders without crashing", () => {
    render(
        <UserProvider>
            <JobCardList jobList={[]} />
        </UserProvider>
    );
})


test("Matches snapshot for no jobs", () => {
    const {asFragment} = render(
        <UserProvider>
            <JobCardList jobList={[]} />
        </UserProvider>
    );

    expect(asFragment()).toMatchSnapshot();
})


test("Matches snapshot for nonempty jobs list", () => {
    const list = [
        {
            id: 0,
            title: "Test title",
            salary: 100000,
            equity: 0,
            companyName: "Test Company"
        },
        {
            id: 1,
            title: "Test title 2",
            salary: 200000,
            equity: 1,
            companyName: "Test Company 2"
        },
    ];

    const {asFragment} = render(
        <UserProvider>
            <JobCardList jobList={list} />
        </UserProvider>
    );

    expect(asFragment()).toMatchSnapshot();
})