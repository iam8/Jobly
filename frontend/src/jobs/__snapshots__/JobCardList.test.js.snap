// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot for no jobs 1`] = `
<DocumentFragment>
  <div
    class="JobCardList"
  />
</DocumentFragment>
`;

exports[`Matches snapshot for nonempty jobs list 1`] = `
<DocumentFragment>
  <div
    class="JobCardList"
  >
    <div
      class="mb-3"
    >
      <div
        class="JobCard"
      >
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <h5
              class="card-title"
            >
              Test title
            </h5>
            <p
              class="text-muted"
            >
              Test Company
            </p>
            <div>
              <small>
                Salary: 100000
              </small>
            </div>
            <div>
              <small>
                Equity: 0
              </small>
            </div>
            <button
              class="float-end btn btn-danger"
              type="button"
            >
              <span>
                Apply
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mb-3"
    >
      <div
        class="JobCard"
      >
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <h5
              class="card-title"
            >
              Test title 2
            </h5>
            <p
              class="text-muted"
            >
              Test Company 2
            </p>
            <div>
              <small>
                Salary: 200000
              </small>
            </div>
            <div>
              <small>
                Equity: 1
              </small>
            </div>
            <button
              class="float-end btn btn-danger"
              type="button"
            >
              <span>
                Apply
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
